anta/tests/snmp.py

```diff
@@ -339,3 +345,85 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"The following SNMP error counters are not found or have non-zero error counters:\n{error_counters_not_ok}")
+
+
+class VerifySnmpUser(AntaTest):
+    """Verifies the SNMP user configurations for specified version(s).
+
+    This test performs the following checks for each specified user:
+
+      1. Verifies that the user name and group name are valid.
+      2. Ensures that the SNMP v3 security model, user authentication, and privacy settings align with version-specific requirements.
+
+    Expected Results
+    ----------------
+    * Success: If all of the following conditions are met:
+        - All specified users are found in the SNMP configuration with valid user groups.
+        - The SNMP v3 security model, user authentication and privacy settings match the required settings.
+    * Failure: If any of the following occur:
+        - A specified user is not found in the SNMP configuration.
+        - A user's group is incorrect.
+        - For SNMP v3 security model, the user authentication and privacy settings do not matches the required settings.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.snmp:
+      - VerifySnmpUser:
+          snmp_users:
+            - username: test
+              group_name: test_group
+              version: v3
+              auth_type: MD5
+              priv_type: AES-128
+    ```
+    """
```

Comment:
```suggestion
    """Verifies the SNMP user configurations.

    This test performs the following checks for each specified user:

      1. User exists in SNMP configuration.
      2. Group assignment is correct.
      3. For SNMPv3 users only:
          - Authentication type matches (if specified)
          - Privacy type matches (if specified)

    Expected Results
    ----------------
    * Success: If all of the following conditions are met:
        - All users exist with correct group assignments.
        - SNMPv3 authentication and privacy types match specified values.
    * Failure: If any of the following occur:
        - User not found in SNMP configuration.
        - Incorrect group assignment.
        - For SNMPv3: Mismatched authentication or privacy types.

    Examples
    --------
    ```yaml
    anta.tests.snmp:
      - VerifySnmpUser:
          snmp_users:
            - username: test
              group_name: test_group
              version: v3
              auth_type: MD5
              priv_type: AES-128
    ```
    """
```

Reply:
Updated. Thanks!!